File: Coding Sample/MS.NET/Fundamentals/Reflection/greeter.cs
public interface IGreeter
{
	string Meet(string name, int age);

	string Leave(string name);
}

public class EnglishGreeter : IGreeter
{
	public string Meet(string name, int age)
	{
		if(age < 17)
			return "Hi " + name;

		return "Hello " + name;
	}

	public string Leave(string name)
	{
		return "Bye " + name;
	}

}

File: Coding Sample/Java/Fundamentals/Reflection/Greeter.java
public interface Greeter{

	String meet(String name, int age);

	String leave(String name);

}

class EnglishGreeter implements Greeter{

	public String meet(String name, int age){
		if(age < 20)
			return "Hi " + name;
		return "Hello " + name;
	}

	public String leave(String name){
		return "Bye " + name;
	}

}

